wayshot/src/wayshot.rs

```diff
@@ -128,41 +129,52 @@ fn main() -> Result<()> {
 
     if let Some(file) = file {
         image_buffer.save(file)?;
-    } else {
+    } else if stdout_print {
         let mut buffer = Cursor::new(Vec::new());
         image_buffer.write_to(&mut buffer, requested_encoding)?;
-
-        if stdout_print {
-            let stdout = stdout();
-            let mut writer = BufWriter::new(stdout.lock());
-            writer.write_all(buffer.get_ref())?;
-        }
+        let stdout = stdout();
+        let mut writer = BufWriter::new(stdout.lock());
+        writer.write_all(buffer.get_ref())?;
         if cli.clipboard {
-            let mut opts = Options::new();
-            match unsafe { fork() } {
-                // Having the image persistently available on the clipboard requires a wayshot process to be alive.
-                // Fork the process with a child detached from the main process and have the parent exit
-                Ok(ForkResult::Parent { .. }) => {
-                    return Ok(());
-                }
-                Ok(ForkResult::Child) => {
-                    opts.foreground(true); // Offer the image till something else is available on the clipboard
-                    opts.copy(
-                        Source::Bytes(buffer.into_inner().into()),
-                        MimeType::Autodetect,
-                    )?;
-                }
-                Err(e) => {
-                    tracing::warn!("Fork failed with error: {e}, couldn't offer image on the clipboard persistently.
-                     Use a clipboard manager to record screenshot.");
-                    opts.copy(
-                        Source::Bytes(buffer.into_inner().into()),
-                        MimeType::Autodetect,
-                    )?;
-                }
-            }
+            clipboard_daemonize(buffer)?;
         }
     }
 
+    if !stdout_print && cli.clipboard {
+        let mut buffer = Cursor::new(Vec::new());
```

Comment:
Do we need the cli.clipboard conditional in the stdout conditional? Is this block outside of the conditional not enough?

```rs

let image_buf: Option<Cursor<Vec<...> = None;
if Some(path) = file {
    /// save
} else {
    /// to stdout

    image_buf = ...;
}

if cli.clipboard && image_buf = Some(image_buffer) {
    //buf stuff
}
```